### PR TITLE
Add first set of tests for APIs with resiliency

### PR DESCRIPTION
**Pull request by halspang**
# Description

This commit adds tests for a series of APIs that are covered by
resiliency including:

- Statestore
- Secrets
- Pubsub
- Bindings
- Actors
- Service invocation

These are generally just unit tests that show things are covered
by a policy/referencing the correct one.

Signed-off-by: Hal Spang <halspang@microsoft.com>

## Issue reference

Please reference the issue this PR will close: Resiliency

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by artursouza — CHANGES_REQUESTED** [review]

File: pkg/actors/actors_mock.go [hunk 1]
```diff
@@ -13,15 +13,16 @@ limitations under the License.
 
 // Code generated by mockery v1.0.0.
 
-package testing
+package actors
```
[the remark this answers is not in the document]
Comment (halspang) [reply pkg/actors/actors_mock.go:16]: @yaron2 - I'm not super familiar with how go packages their binaries. Is this a risk here? If it is, can I use build tags like the e2e tests? Adding something like `// build:test` to the top of the file?

[the remark this answers is not in the document]
Comment (berndverst) [reply pkg/actors/actors_mock.go:16]: Yes this code would be included in the production binary.

If this package name is required build tags like `// +build unit` and then running test with `go test -v --tags=unit` seems like the way to go.

**Review by berndverst — APPROVED** [review]
Assuming use of build tags or moving new test code to `_test.go` files or `testing` package this looks good.

**Comment by halspang** [comment]
Added a build tag @artursouza. The alternate here I guess would just be redefining the testing code inside of `actors_test.go`. I think the build tag is the more correct solution though.

**Review by artursouza — APPROVED** [review]

**Pull request merged**